README.md

```diff
@@ -6,53 +6,143 @@
+
+1. Run `bin/tapioca generate`. This generates RBIs for the gems in your application
+2. Try to run the type checker (`bundle exec srb tc`)
+3. If you notice that definitions are missing for gems, you might need to add the gem for which those definitions belong to in `sorbet/tapioca/require.rb`. Before manually adding requires, try running `bin/tapioca require` so that Tapioca can figure out the requires. If this still does not work, proceed to write manual requires
+4. After requiring the gems inside that file, go back to 1. until there are no more errors coming from gems
+
+### 6. Run Tapioca DSL
+
+With all the gem RBIs in place, now Tapioca can generate DSL RBIs. DSL RBIs are definitions that only exist in runtime. For example,
+```rb
+class Post < ApplicationRecord
+  # Belongs to will create a few methods in this class
+  # to be able to access the associated author. Sorbet
+  # does not know about them, since they only exist during
+  # runtime. Tapioca can generate the definitions in RBIs
+  # for methods like this one.
+  belongs_to :author
 end
 ```
+Run `bin/tapioca dsl` to generate the runtime definitions for your application.
 
-and do not forget to execute `tapioca` using `bundler`:
+While going through steps 5. and 6., you might realize that some shims have become obsolete or incorrect. Make sure to edit or remove existing shims that are causing errors.
+
+### 7. Run Tapioca todo
+
+After steps 5. and 6., if there are still errors for missing definitions, you may want to try bringing back the shims you saved for later.
+
+If the previous shims do not satisfy all errors, you can also skip fixing them for later with the todo command. Running `bin/tapioca todo` will generate a file with empty definitions, and allows developers to track which definitions are still missing. This is optional, you might want to actually write the complete signatures in shims.
+
+### 8. Bump strictness on files
+
+After all the RBIs are generated and no type errors are occurring, some files might actually be ready to move from `typed: false` to `typed: true`. [Spoom](https://github.com/Shopify/spoom) can automatically bump files to `true` if doing so produces no new typing errors.
+
+To bump all possible files to `true`, simply run `bundle exec spoom bump`.
+
+### Things that do not need to happen
+
+Here are some things that do not need to be run for the migration (or ever in some cases).
+
+1. `bin/tapioca sync`: this command is meant to be used **after** the initial setup is complete to update RBIs for gems that have been upgraded. It does not need to be used as part of the migration process, but should be used later
+2. `bundle exec srb rbi hidden-definitions`: not necessary at all, ever
+3. `bundle exec srb rbi suggest-typed`: not necessary at all, ever. Prefer `bundle exec spoom bump`
+
+Done! Your application should be all set and type checking should pass.
+
+## How does tapioca compare to "srb rbi gems" ?
```

Comment:
If we're going to document all of this here, we should also add a section explaining how to keep the RBIs up-to-date when the Gemfile changes and the files related to DSLs change.